specification/resourcemover/resource-manager/Microsoft.Migrate/stable/2021-01-01/resourcemovercollection.json

```diff
@@ -1852,526 +1951,13 @@
             "read"
           ]
         },
-        "targetResourceName": {
-          "description": "Gets or sets the target Resource name.",
+        "serializedResourceSettings": {
+          "description": "Gets or sets the resource provider specific serialized resource settings. Resource providers need to document the settings.",
           "type": "string"
         }
       },
       "discriminator": "resourceType"
     },
-    "VirtualMachineResourceSettings": {
```

Comment:
1. Shouldn't these stay to support inter operability? 
2. Considering above, should we introduce a new derived class called GenericResourceSettings?